### PR TITLE
[PotentialFlow] Avoiding compilation warning in define wake process.

### PR DESCRIPTION
**Pull request by marcnunezc**
Adressing #5449

### CODE REVIEW

File: applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp [hunk 1]
```diff
@@ -119,16 +119,19 @@ void Define2DWakeProcess::SetWakeDirectionAndNormal()
 // This function finds and saves the trailing edge for further computations
 void Define2DWakeProcess::SaveTrailingEdgeNode()
 {
+    KRATOS_ERROR_IF(mrBodyModelPart.NumberOfNodes() == 0) << "There are no nodes in the body_model_part!"<< std::endl;
+
     double max_x_coordinate = std::numeric_limits<double>::lowest();
-    NodeType* trailing_edge_node;
+    NodeType& trailing_edge_node = mrBodyModelPart.GetNode(1);
```
Comment (rubenzorrilla) [inline applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp]: I think you can save the reference to this auxiliary node by just saving the `Id`. Then you can do `mpTrailingEdgeNode = &mrBodyModelPart.GetNode(aux_id);` and use the member pointer to do the `mpTrailingEdgeNode->SetValue(TRAILING_EDGE,true); `. I think that this makes also possible to have a `const` iterator in the loop.

**Comment by philbucher** [comment]
I have a better suggestion will write soon

File: applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp [hunk 2]
```diff
@@ -119,16 +119,19 @@ void Define2DWakeProcess::SetWakeDirectionAndNormal()
...
         if (r_node.X() > max_x_coordinate) {
             max_x_coordinate = r_node.X();
-            trailing_edge_node = &r_node;
+            trailing_edge_node = r_node;
         }
     }
-    trailing_edge_node->SetValue(TRAILING_EDGE, true);
-    mpTrailingEdgeNode = trailing_edge_node;
+    trailing_edge_node.SetValue(TRAILING_EDGE, true);
+    mpTrailingEdgeNode = &trailing_edge_node;
```
Comment (philbucher) [inline applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp]: I would do the following:
~~~c++
KRATOS_ERROR_IF(mrBodyModelPart.NumberOfNodes() == 0) << "There are no nodes in the body_model_part!"<< std::endl;

double max_x_coordinate = std::numeric_limits<double>::lowest();
NodeType* p_trailing_edge_node = mrBodyModelPart.NodesBegin();

for (const auto& r_node : mrBodyModelPart.Nodes()) {
    if (r_node.X() > max_x_coordinate) {
        max_x_coordinate = r_node.X();
        p_trailing_edge_node = *r_node;
    }
}
p_trailing_edge_node->SetValue(TRAILING_EDGE, true);
mpTrailingEdgeNode = p_trailing_edge_node;
~~~

this way you don't have to deal with aux-ids

Comment (marcnunezc) [reply applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp]: Is this correct?:

`NodeType* p_trailing_edge_node = mrBodyModelPart.NodesBegin();`

It gives the following error: 

```
/home/marc/Kratos/applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp: In member function ‘void Kratos::Define2DWakeProcess::SaveTrailingEdgeNode()’:
/home/marc/Kratos/applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp:125:65: error: cannot convert ‘Kratos::ModelPart::NodeIterator {aka boost::iterators::indirect_iterator<__gnu_cxx::__normal_iterator<Kratos::intrusive_ptr<Kratos::Node<3> >*, std::vector<Kratos::intrusive_ptr<Kratos::Node<3> >, std::allocator<Kratos::intrusive_ptr<Kratos::Node<3> > > > >, boost::iterators::use_default, boost::iterators::use_default, boost::iterators::use_default, boost::iterators::use_default>}’ to ‘Kratos::Define2DWakeProcess::NodeType* {aka Kratos::Node<3>*}’ in initialization
     NodeType* p_trailing_edge_node = mrBodyModelPart.NodesBegin();

```

`NodesBegin()` is an iterator right? Can that be converted to a pointer?

In any case, I pushed @rubenzorrilla suggestion, do you like it?

Comment (philbucher) [reply applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp]: > Can that be converted to a pointer?

Of course :)
I don't know how by hand, but I suggest either google or playing with `*`

Comment (philbucher) [reply applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp]: > In any case, I pushed @rubenzorrilla suggestion, do you like it?

I don't, you introduced an unnecessary find by not saving the node right away

Comment (marcnunezc) [reply applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp]: Ok let me check it then!

Comment (marcnunezc) [reply applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp]: So the operator was: 
`auto p_trailing_edge_node = &*mrBodyModelPart.NodesBegin();`

**Review by philbucher — APPROVED** [review]
👍

**Pull request merged**